Assets/Scripts/Environment/Hangar/Battery.cs

```diff
@@ -55,38 +62,77 @@ private void SetEmission()
         meshRenderer.material.SetColor("_EmissionColor", emission);
     }
 
-    public bool PlayerIsHolding()
+    /*public bool PlayerIsHolding()
     {
-        return PlayerManager.PlayerObject.GetComponentInChildren<Battery>() != null;
+        return springJoint != null && springJoint.connectedBody == PlayerManager.PlayerMovement.PlayerRigidbody;
+
+        return PlayerManager.IsHoldingBattery;
     }
+    */
 
     public void TakeBattery()
     {
-        Container.ParentToPlayer();
+        Container.transform.localPosition = new Vector3(Container.transform.localPosition.x, HangarConstants.BatteryYPosition, Container.transform.localPosition.z);
+
+        containerRigidBody.useGravity = false;
+
+        // We add constraints so that the battery doesn't swing around uncontrollably
+        containerRigidBody.constraints = HangarConstants.BatteryRigidbodyConstraints;
+
+        // Setting the container's spring joint values
+        springJoint = Container.AddComponent<SpringJoint>();
+        springJoint.connectedBody = PlayerManager.PlayerObject.GetComponent<Rigidbody>();
+
+        springJoint.spring = HangarConstants.Spring;
+        springJoint.damper = HangarConstants.Damper;
+        springJoint.minDistance = HangarConstants.MinDistance;
+        springJoint.maxDistance = HangarConstants.MaxDistance;
+        springJoint.tolerance = HangarConstants.Tolerance;
+        springJoint.enableCollision = HangarConstants.EnableCollision;
+        springJoint.breakForce = HangarConstants.BreakForce;
+
+        PlayerManager.IsHoldingBattery = true;
+
+        springJointExists = true;

```

Comment:
I don't think we need this, if we use the PlayerIsHolding method?

If we do need it, I think it would be better to make a property for it, rather than set it to true/false each time. Because I think it's just equivalent to a null check? 

`private bool SpringJointExists => springJoint != null;`